tools/merge.py

```diff
@@ -0,0 +1,441 @@
+                elif module_name == "mp_world_size":
+                    # overwrite mp in sd
+                    out_sd[module_name] = model_parallel_size
+                elif module_name == "dp_world_size":
+                    out_sd[module_name] = int(
+                        partitions[0] * (model_parallel_size / orig_mp)
+                    )
+                elif module_name == "args":
+                    # change mp size in sd args
+                    p = partitions[0]
+                    replace(p, "model-parallel-size", model_parallel_size)
+                else:
+                    out_sd[module_name] = partitions[0]
+            else:
+                partition_dim = PARTITION_DIM_MAP.get(module_name, None)
+                if partition_dim is None:
+                    print(module_name)
+                    # just take the 0th partition for non model-parallel weights
+                    if parameter_mean:
+                        out_sd[module_name] = torch.mean(torch.stack(partitions), dim=0)
+                    else:
+                        out_sd[module_name] = partitions[0]
+                else:
+                    splits = merge_partitions(
+                        partitions, partition_dim=partition_dim, mp=model_parallel_size
+                    )
+                    for mp_rank in range(model_parallel_size):
+                        split_modules[module_name].append(splits[mp_rank])
+
+        # save outputs:
+        if layer == "model_states":
+            pbar.set_description(
+                f"* Saving state dicts for layer {layer}, module `{module_name}`"
+            )
+            for pp in range(pipe_parallel_size):
+                out_path = output_weights_dir / f"mp_rank_{pp:02}_model_states.pt"
+                torch.save(out_sd, out_path)
+            pbar.set_description(
+                f"* Saved state dicts for layer {layer}, module `{module_name}`"
+            )
+        else:
+            for mp_rank in range(model_parallel_size):
+                out_path = (
+                    output_weights_dir
+                    / f"layer_{layer}-model_{mp_rank:02}-model_states.pt"
+                )
+                for module_name, splits in split_modules.items():
+                    out_sd[module_name] = splits[mp_rank]
+                torch.save(out_sd, out_path)
+                pbar.set_description(
+                    f"* Saved state dicts for layer {layer}, module `{module_name}`, and mp rank {mp_rank}"
+                )
+
+    # write 'latest' file
+    with open(output_dir / "latest", "w") as f:
+        f.write(f"global_step{global_step}")
+    print("* DONE!")
+
+
+if __name__ == "__main__":
+    # get arguments
+    args = parse_args()
+    merge_checkpoints(
+        args.checkpoint_dir,
+        args.model_parallel,
+        args.pipe_parallel,
+        args.output_dir,
+        args.global_step,
+        args.layernorm_mean,
```

Comment:
this line was throwing an error for me because it's not on the args list

Reply:
Should be `parameter_mean`